tracer/src/Datadog.Trace.Tools.dd_dotnet/Checks/ProcessBasicCheck.cs

```diff
@@ -0,0 +1,463 @@
+            }
+            else
+            {
+                Utils.WriteWarning(runtime == ProcessInfo.Runtime.Mixed ? BothRuntimesDetected : RuntimeDetectionFailed);
+                runtime = ProcessInfo.Runtime.NetFx;
+            }
+
+            bool isContinuousProfilerEnabled;
+
+            if (process.EnvironmentVariables.TryGetValue("DD_PROFILING_ENABLED", out var profilingEnabled))
+            {
+                if (ParseBooleanConfigurationValue(profilingEnabled))
+                {
+                    AnsiConsole.WriteLine(ContinuousProfilerEnabled);
+                    isContinuousProfilerEnabled = true;
+                }
+                else
+                {
+                    AnsiConsole.WriteLine(ContinuousProfilerDisabled);
+                    isContinuousProfilerEnabled = false;
+                }
+            }
+            else
+            {
+                AnsiConsole.WriteLine(ContinuousProfilerNotSet);
+                isContinuousProfilerEnabled = false;
+            }
+
+            var loaderModule = FindLoader(process);
+            var nativeTracerModule = FindNativeTracerModule(process, loaderModule != null);
+
+            if (loaderModule == null)
+            {
+                AnsiConsole.WriteLine(LoaderNotLoaded);
+            }
+
+            if (nativeTracerModule == null)
+            {
+                Utils.WriteWarning(NativeTracerNotLoaded);
+                ok = false;
+            }
+            else
+            {
+                // Only check the version of the native binary on Windows
+                // .so modules don't have version metadata
+                if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+                {
+                    if (!Version.TryParse(FileVersionInfo.GetVersionInfo(nativeTracerModule).FileVersion, out nativeTracerVersion))
+                    {
+                        nativeTracerVersion = null;
+                    }
+
+                    AnsiConsole.WriteLine(ProfilerVersion(nativeTracerVersion != null ? $"{nativeTracerVersion}" : "{empty}"));
+                }
+            }
+
+            if (isContinuousProfilerEnabled)
+            {
+                ok &= CheckContinousProfiler(process, loaderModule);
+            }
+
+            var tracerModules = FindTracerModules(process).ToArray();
+
+            if (tracerModules.Length == 0)
+            {
+                Utils.WriteWarning(TracerNotLoaded);
+                ok = false;
+            }
+            else if (tracerModules.Length == 1)
+            {
+                var version = FileVersionInfo.GetVersionInfo(tracerModules[0]);
+                AnsiConsole.WriteLine(TracerVersion(version.FileVersion ?? "{empty}"));
+            }
+            else if (tracerModules.Length > 1)
+            {
+                // There are too many tracers in there. Find out if it's bad or very bad
+                bool areAllVersion2 = true;
```

Comment:
Given we're talking about a v3, I wonder how many places we've hardcoded things like this 🤔